AudioSystem/AudioSystemExtension.cs

```diff
@@ -159,55 +159,66 @@ public static AudioSource Play(this AudioConfig audioConfig, float volume, Trans
+				onComplete, onEachConfigPlayed, onTimerStarted);
 		}
 
-		public static AudioSource PlaySequence(this AudioConfigSequence compositeConfig, float volume = 1f, Transform parent = null, 
-			Action onComplete = null, Action<int> onEachConfigPlayed = null)
+		public static AudioConfig PlaySequence(this AudioConfigSequence compositeConfig, float volume = 1f, Transform parent = null, 
+			Action onComplete = null, Action<AudioConfig, int> onEachConfigPlayed = null, Action<Timer> onTimerStarted = null)
 		{
-			return PlaySequence(compositeConfig.GetAudioConfigs(), volume, parent, compositeConfig.GetClipIndex, compositeConfig.GetPostClipDelay, onComplete, onEachConfigPlayed);
+			return PlaySequence(compositeConfig.AudioConfigs, volume, parent, compositeConfig.GetClipIndex, compositeConfig.GetClipDelay, 
+				onComplete, onEachConfigPlayed, onTimerStarted);
 		}
 
-		public static AudioSource PlaySequence(this AudioConfigSequence compositeConfig, Action onComplete = null)
+		public static AudioConfig PlaySequence(this AudioConfigSequence compositeConfig, Action onComplete = null)
 		{
 			return PlaySequence(compositeConfig, onComplete: onComplete);
 		}
 
-		private static AudioSource PlaySequenceAtIndex(int index, IEnumerator<AudioConfig> enumerator, float volume, Transform parent = null, 
-			Func<int, int> getClipId = null, Func<int, float> getClipDelay = null, Action onComplete = null, Action<int> onEachConfigPlayed = null)
+		private static AudioConfig PlaySequenceAtIndex(AudioConfig[] configs, int index, float volume, Transform parent = null, 
+			Func<int, int> getClipId = null, Func<int, float> getClipDelay = null, 
+			Action onComplete = null, Action<AudioConfig, int> onEachConfigPlayed = null, Action<Timer> onTimerStarted = null)
 		{
-			Func<AudioSource> playNextConfig = () =>
+			Func<AudioConfig> playThisConfig = () =>
 			{
-				if (!enumerator.MoveNext())
+				if (index >= configs.Length)
 				{
 					onComplete.SafeInvoke();
 					return null;
 				}
 
-				onEachConfigPlayed.SafeInvoke(index);
+				var thisConfig = configs[index];
 
-				return enumerator.Current.Play(volume, parent, (getClipId != null) ? getClipId(index) : AudioConfig.RANDOM_CLIP, () =>
+				onEachConfigPlayed.SafeInvoke(thisConfig, index);
+
+				thisConfig.Play(volume, parent, (getClipId != null) ? getClipId(index) : AudioConfig.RANDOM_CLIP, () =>
 				{
-					PlaySequenceAtIndex(index + 1, enumerator, volume, parent, getClipId, getClipDelay, onComplete, onEachConfigPlayed);
+					PlaySequenceAtIndex(configs, index + 1, volume, parent, getClipId, getClipDelay, onComplete, onEachConfigPlayed);
 				});
+
+				return thisConfig;
 			};
 
 			var delay = (getClipDelay != null) ? getClipDelay(index) : 0f;
 
 			if (index == 0 || delay <= 0f)
 			{
-				return playNextConfig();
+				return playThisConfig();
 			}
 			else
 			{
-				Timer.SetTimeout(delay, () => playNextConfig());
+				var timer = Timer.SetTimeout(delay, () => playThisConfig());
```

Comment:
I think you can pass in the function straight away?

like `var timer = Timer.SetTimeout(delay, playThisConfig);`

Reply:
Apparently not